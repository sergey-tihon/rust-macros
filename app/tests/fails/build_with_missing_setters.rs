use builder_macro::Builder;

#[derive(Builder)]
struct Gleipnir {
    roots_of: String,
    breath_of_a_fish: u8,
    anything_else: bool,
}
fn main() {
    Gleipnir::builder()
        .roots_of("mountains".to_string())
        .breath_of_a_fish(1)
        // missing final property
        .build();
}
