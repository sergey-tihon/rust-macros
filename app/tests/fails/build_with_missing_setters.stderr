error[E0599]: no method named `build` found for struct `GleipnirBuilder<__AnythingElseOfGleipnirBuilder>` in the current scope
  --> tests/fails/build_with_missing_setters.rs:14:10
   |
3  |   #[derive(Builder)]
   |            ------- method `build` not found for this struct
...
10 | /     Gleipnir::builder()
11 | |         .roots_of("mountains".to_string())
12 | |         .breath_of_a_fish(1)
13 | |         // missing final property
14 | |         .build();
   | |         -^^^^^ method not found in `GleipnirBuilder<__AnythingElseOfGleipnirBuilder>`
   | |_________|
   |
   |
   = note: the method was found for
           - `GleipnirBuilder<FinalBuilder>`
